private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go

```diff
@@ -476,10 +476,21 @@ func addFieldChangedType(add addFunc, previousField protosource.Field, field pro
 	}
 	// otherwise prints as hex
 	previousNumberString := strconv.FormatInt(int64(previousField.Number()), 10)
+
+	var typeNameLocation protosource.Location
+
+	// TypeLocation does not work on message types for some reason.
+	switch field.Type() {
+	case protosource.FieldDescriptorProtoTypeMessage:
+		typeNameLocation = field.Location()
```

Comment:
I think you want `TypeNameLocation()` here. Also, this applies if the type is a group or enum, too.